// @clerk/nextjs/dist/app-beta/client/ui-components.js
"use strict";
'use client';
Object.defineProperty(exports, "__esModule", { value: true });
exports.OrganizationSwitcher = exports.OrganizationProfile = exports.CreateOrganization = exports.UserProfile = exports.UserButton = exports.SignUp = exports.SignIn = void 0;
const tslib_1 = require("tslib");
const clerk_react_1 = require("@clerk/clerk-react");
Object.defineProperty(exports, "CreateOrganization", { enumerable: true, get: function () { return clerk_react_1.CreateOrganization; } });
Object.defineProperty(exports, "OrganizationProfile", { enumerable: true, get: function () { return clerk_react_1.OrganizationProfile; } });
Object.defineProperty(exports, "OrganizationSwitcher", { enumerable: true, get: function () { return clerk_react_1.OrganizationSwitcher; } });
Object.defineProperty(exports, "UserProfile", { enumerable: true, get: function () { return clerk_react_1.UserProfile; } });
const navigation_1 = require("next/navigation");
const react_1 = tslib_1.__importDefault(require("react"));
const UserButton = (props) => {
    const pathname = (0, navigation_1.usePathname)();
    const searchParams = (0, navigation_1.useSearchParams)();
    return (react_1.default.createElement(clerk_react_1.UserButton, { afterSignOutUrl: props.afterSignOutUrl || `${pathname}${searchParams.toString()}`, ...props }));
};
exports.UserButton = UserButton;
const SignIn = (props) => {
    const pathname = (0, navigation_1.usePathname)();
    const searchParams = (0, navigation_1.useSearchParams)();
    return (react_1.default.createElement(clerk_react_1.SignIn, { redirectUrl: props.redirectUrl || `${pathname}${searchParams.toString()}`, ...props }));
};
exports.SignIn = SignIn;
const SignUp = (props) => {
    const pathname = (0, navigation_1.usePathname)();
    const searchParams = (0, navigation_1.useSearchParams)();
    return (react_1.default.createElement(clerk_react_1.SignUp, { redirectUrl: props.redirectUrl || `${pathname}${searchParams.toString()}`, ...props }));
};
exports.SignUp = SignUp;
